airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,326 @@
+            )
+
+        # TODO: is there a fast way to check if the connector is already using the latest CDK version?
+        # this is better than all the docker shenanigans if it's not needed.
+        # probably just grep the toml file. If can't find it fail over to the better way
+
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+class UpdateCdk(Step):
+    context: ConnectorContext
+
+    title = "Update the CDK to a new version."
+
+    def __init__(self, context: PipelineContext, cdk_version: str) -> None:
+        super().__init__(context)
+        self.cdk_version = cdk_version
+
+    async def _run(self) -> StepResult:
+        base_image_name = self.context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        base_container = self.dagger_client.container(platform=LOCAL_BUILD_PLATFORM).from_(base_image_name)
+        connector_container = await with_python_connector_installed(
+            self.context,
+            base_container,
+            str(self.context.connector.code_directory),
+        )
+
+        try:
+            poetry_show_result = await connector_container.with_exec(sh_dash_c(["poetry show | grep airbyte-cdk"])).stdout()
+            if not poetry_show_result:
+                return StepResult(step=self, status=StepStatus.FAILURE, stderr="CDK is not installed in the connector.")
+            current_cdk_version = extract_poetry_show_version(poetry_show_result)
+            if not current_cdk_version:
+                return StepResult(step=self, status=StepStatus.FAILURE, stderr="CDK was not found in the connector poetry.")
+
+            self.logger.info(f"Current CDK version for {self.context.connector.technical_name}: {current_cdk_version}")
+
+            # TODO: maybe add --force to also downgrade the CDK version
+            if compare_semver_versions(current_cdk_version, self.cdk_version) >= 0:
+                return StepResult(
+                    step=self,
+                    status=StepStatus.SKIPPED,
+                    stderr=f"CDK is already up to date. Current version: {current_cdk_version}, requested version: {self.cdk_version}",
+                )
+
+            with_new_cdk = await connector_container.with_exec(["poetry", "add", f"airbyte-cdk=={self.cdk_version}"])
+            await with_new_cdk.file("pyproject.toml").export(f"{self.context.connector.code_directory}/pyproject.toml")
+            self.logger.info(f"Generated pyproject.toml for {self.context.connector.technical_name}")
+            await with_new_cdk.file("poetry.lock").export(f"{self.context.connector.code_directory}/poetry.lock")
+            self.logger.info(f"Generated poetry.lock for {self.context.connector.technical_name}")
+        except dagger.ExecError as e:
+            return StepResult(step=self, status=StepStatus.FAILURE, stderr=str(e))
+
+        return StepResult(step=self, status=StepStatus.SUCCESS)
+
+
+class RestoreOriginalState(Step):
```

Comment:
It's not obvious to me why this is needed if we're just leaving outstanding changes around that will be blown away anyway. Personally I'd just use `git reset --hard head`.

Reply:
I followed the pattern from other ones. I've been running these commands lcoally in tandem with other things and it's been pretty helpful to clean themselves up so we know what's truly changed and can be commited. In that world, the `reset` would be dangerous, too.

Reply:
Ah right. Let's leave it.